# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFontUtils.h
# Commit: 8b26219a4294
# Full Hash: 8b26219a4294a083f2d22da23bd8958234e4dd1b
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-08-16 16:01:29
# Regressor Bug: 1740530
# File Overlap Count: 1
# Description:
#   Bug 1740530 - patch 1 - Move COLR font support functions from gfxUtils into their own class. r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152036
# ==============================================================================

diff -r 8d99a5abccc7 -r 8b26219a4294 gfx/thebes/gfxFontUtils.h
--- a/gfx/thebes/gfxFontUtils.h	Tue Aug 16 06:00:43 2022 -0400
+++ b/gfx/thebes/gfxFontUtils.h	Tue Aug 16 10:39:50 2022 +0000
@@ -31,9 +31,6 @@
 namespace mozilla {
 class Encoding;
 class ServoStyleSet;
-namespace gfx {
-struct DeviceColor;
-}
 }  // namespace mozilla
 
 /* Bug 341128 - w32api defines min/max which causes problems with <bitset> */
@@ -790,22 +787,6 @@
   AutoSwap_PRUint16 tupleIndex;
 };
 
-struct COLRHeader {
-  AutoSwap_PRUint16 version;
-  AutoSwap_PRUint16 numBaseGlyphRecord;
-  AutoSwap_PRUint32 offsetBaseGlyphRecord;
-  AutoSwap_PRUint32 offsetLayerRecord;
-  AutoSwap_PRUint16 numLayerRecords;
-};
-
-struct CPALHeaderVersion0 {
-  AutoSwap_PRUint16 version;
-  AutoSwap_PRUint16 numPaletteEntries;
-  AutoSwap_PRUint16 numPalettes;
-  AutoSwap_PRUint16 numColorRecords;
-  AutoSwap_PRUint32 offsetFirstColorRecord;
-};
-
 #pragma pack()
 
 // Return just the highest bit of the given value, i.e., the highest
@@ -1151,15 +1132,6 @@
   // generate a unique font name
   static nsresult MakeUniqueUserFontName(nsAString& aName);
 
-  // for color layer from glyph using COLR and CPAL tables
-  static bool ValidateColorGlyphs(hb_blob_t* aCOLR, hb_blob_t* aCPAL);
-  static bool GetColorGlyphLayers(
-      hb_blob_t* aCOLR, hb_blob_t* aCPAL, uint32_t aGlyphId,
-      const mozilla::gfx::DeviceColor& aDefaultColor,
-      nsTArray<uint16_t>& aGlyphs,
-      nsTArray<mozilla::gfx::DeviceColor>& aColors);
-  static bool HasColorLayersForGlyph(hb_blob_t* aCOLR, uint32_t aGlyphId);
-
   // Helper used to implement gfxFontEntry::GetVariation{Axes,Instances} for
   // platforms where the native font APIs don't provide the info we want
   // in a convenient form, or when native APIs are too expensive.